.changeset/grumpy-eyes-cross.md

```
@@ -0,0 +1,13 @@
+---
+'@mastra/server': minor
+---
+
+Added a framework guarantee that user-registered server middleware can no longer block routes declared public via `createPublicRoute()` or `requiresAuth: false`.
```

markdownlint-cli2 (0.23.2)

[warning] First line in a file should be a top-level heading

(MD041, first-line-heading, first-line-h1)